### PR TITLE
PS-4487: `-DWITH_NUMA=ON` is silently ignored when libnuma-dev is not installed

### PR DESCRIPTION
**Pull request by inikep**
1. cmake with `-DWITH_NUMA=ON` will return error when libnuma-dev is not installed
2. Fix `sys_vars.innodb_numa_interleave_basic` that showed `Test requires: NUMA must be enabled`
   even when PS was compiled with `-DWITH_NUMA=ON`

Backported changes from 5.7 basing on:
https://github.com/percona/percona-server/commit/b3c011f0b26bad5a1d4fc381ca241b88b07e4767
https://github.com/percona/percona-server/commit/3fc9d6d1780540e7f50d7ff6323c3c873f5038c6
https://github.com/percona/percona-server/commit/de921a8185c19f64e3d8a68fcb148c5773f567ee
https://github.com/percona/percona-server/commit/36ebe7d9af9146cdb194bf7f50577ee2fe945296

### CODE REVIEW

**Comment by inikep** [comment]
https://jenkins.percona.com/job/percona-server-5.6-param/2182/

**Review by laurynas-biveinis — APPROVED** [review]
* I am not sure InnoDB source changes have to be backported. On one hand, they add to our diff against 5.6 with no good reason, on another, reduce the diff against the same in 5.7.
* Besides Jenkins, has this been tested w/&w/o -DWITH_NUMA=ON on systems w/&w/o libnuma-dev, to see that the right thing happens in all four cases?

**Comment by inikep** [comment]
I verified locally all cases with `-DWITH_NUMA=ON`, `-DWITH_NUMA=OFF`, and without `-DWITH_NUMA`. All works as expected.

File: storage/innobase/buf/buf0buf.cc [hunk 1]
```diff
@@ -53,10 +53,13 @@ Created 11/5/1995 Heikki Tuuri
...
 #include "srv0mon.h"
 #include "buf0checksum.h"
+
+UNIV_INTERN my_bool  srv_numa_interleave = FALSE;
+
 #ifdef HAVE_LIBNUMA
 #include <numa.h>
 #include <numaif.h>
-#endif // HAVE_LIBNUMA
+#endif /* HAVE_LIBNUMA */
```
Comment (percona-ysorokin) [inline storage/innobase/buf/buf0buf.cc]: Doesn't look like a meaningful change. I would leave this as is.

File: storage/innobase/buf/buf0buf.cc [hunk 2]
```diff
@@ -1167,7 +1170,7 @@ buf_chunk_init(
 				" (error: %s).", strerror(errno));
 		}
 	}
-#endif // HAVE_LIBNUMA
+#endif /* HAVE_LIBNUMA */
```
Comment (percona-ysorokin) [inline storage/innobase/buf/buf0buf.cc]: Likewise

File: storage/innobase/handler/ha_innodb.cc [hunk 3]
```diff
@@ -18535,7 +18535,7 @@ static MYSQL_SYSVAR_BOOL(numa_interleave, srv_numa_interleave,
   PLUGIN_VAR_NOCMDARG | PLUGIN_VAR_READONLY,
   "Use NUMA interleave memory policy to allocate InnoDB buffer pool.",
   NULL, NULL, FALSE);
-#endif // HAVE_LIBNUMA
+#endif /* HAVE_LIBNUMA */
```
Comment (percona-ysorokin) [inline storage/innobase/handler/ha_innodb.cc]: Likewise

File: storage/innobase/handler/ha_innodb.cc [hunk 4]
```diff
@@ -18903,7 +18903,7 @@ static struct st_mysql_sys_var* innobase_system_variables[]= {
   MYSQL_SYSVAR(use_native_aio),
 #ifdef HAVE_LIBNUMA
   MYSQL_SYSVAR(numa_interleave),
-#endif // HAVE_LIBNUMA
+#endif /* HAVE_LIBNUMA */
```
Comment (percona-ysorokin) [inline storage/innobase/handler/ha_innodb.cc]: Kilewise

**Pull request merged**